vm/ieleClientUtils.ml

```diff
@@ -19,16 +19,19 @@ let of_hex signed str =
 let of_hex_unsigned = of_hex false
 let of_hex = of_hex true
 
-let to_hex bytes =
-  if Bytes.length bytes = 0 then "0x00" else
+let to_hex signed bytes =
```

Comment:
I think it would be better to just have separate functions here,
`let to_hex_unsigned bytes = "0x" ^ (match Hex.of_string (Bytes.to_string bytes) with (`Hex str) -> str)`
is simpler than trying to combine the cases.